### ISSUE TITLE
`validate` false positive when property name is `Object` method

### ISSUE THREAD

**Issue by smathy** [apotonick/reform#314]
Might not be only `Object` methods, but is definitely at least those:

```ruby
require "reform/form/active_model/validations"
class BadForm < Reform::Form
  include Reform::Form::ActiveModel::Validations
  
  property :trust        # or anything from `Object.public_methods`
  validates :trust, presence: true
end

BadForm.new(Struct.new(:trust).new).validate({})
# => true
```

**Comment by apotonick** [comment]
How could we catch that, though??

We can't test if the method is already defined: while this might work in a normal form, in inherited forms, this will create warnings even though nothing is wrong. Any ideas?

**Comment by smathy** [comment]
I don't know how you've implemented that stuff, it seems that instead of using some internal structure for the properties (eg. hash keys) that at the time of validation those properties have already been attached to a ruby object as methods, which means there will always be a false positive for any existing `Object` methods.

I think the validation needs to be done on a separate data structure where the properties are not methods.

**Comment by apotonick** [comment]
The validation _is_ done in another object, which is still an object, though! :grimacing: 

The problem is when initializing the form, the form asks the model for its properties, e.g. `model.trust` and that - apparently - returns true?

**Comment by smathy** [comment]
Right, maybe it should ask `model.attributes` or something, because based on what I think you're saying even an AR method, our method in the model itself will cause a false positive

**Comment by apotonick** [comment]
Does your model have a `trust` field? Can you paste results for the following test?

```ruby
model.trust
YourForm.new(model).trust
```

**Comment by smathy** [comment]
You mean in the original code, or in the reproduction that I included in this issue?  Obviously there the model is just `Struct.new(:trust).new` so your example becomes:

```ruby
[29] pry(main)> model = Struct.new(:trust).new
=> #<struct  trust=nil>
[30] pry(main)> model.trust
=> nil
[31] pry(main)> BadForm.new(model).trust
=> nil
```

**Comment by smathy** [comment]
...and yet:

```ruby
[32] pry(main)> BadForm.new(Struct.new(:trust).new).validate({})
=> true
```

**Comment by RKushnir** [comment]
I think, the problem essentially boils down to https://github.com/apotonick/reform/blob/79032a3b86378306b3942ab1948e42168697c66c/lib/reform/form/active_model/validations.rb#L68
```
> model = Struct.new(:trust).new
=> #<struct  trust=nil>
> SimpleDelegator.new(model).trust
=> #<struct  trust=nil>
> model.trust
=> nil
```

**Comment by fran-worley** [comment]
As of Reform 2.2 all active model / active record related code has moved to the reform-rails gem.

If you're still experiencing this issue, please can you reopen it in the correct gem. Thanks 😺

**Comment by smathy** [comment]
Hmm, working minimal code showing problem, months pass, then redirect to another gem without even a link.  Nope.

If you're interested in seeing if this bug still exists in your gem then paste my sample code into a rails console and you'll see.

**Comment by fran-worley** [comment]
Thanks for coming back to me @smathy. The reform- rails gem can be found at https://github.com/trailblazer/reform-rails. I didn't include the link in the first place as its the first and only repo of that name that comes up when you search github for reform-rails.

The main reason I closed this was that there was no update from you since 2015 and the bump in June a) triggered no response from yourself and b) is specific to ActiveModel Validations which has been moved to the new gem as I mentioned.

Are you still using Reform? It would be useful to know if the problem persists in 2.2.
